sentinel3: error on LST + solar angles

Getting error:
```
E                   reprojected_data = apply_LUT_on_band(band_data, LUT_angles, band_settings.get('_FillValue', None))  # result is an numpy array with reprojected data
E                 File "/opt/openeo/lib/python3.8/site-packages/openeogeotrellis/collections/sentinel3.py", line 638, in apply_LUT_on_band
E                   grid_values = data_flat[LUT]
E               IndexError: index 50018 is out of bounds for axis 0 with size 46743
```

On this load_collection:
```
datacube = con.load_collection(
        "SENTINEL3_SLSTR_L2_LST",
        spatial_extent={"west": -95, "south": 40, "east": -93, "north": 44},
        temporal_extent=["2021-08-03", "2021-08-04"],
        bands=["sunZenithAngles"]
    )
```

--- a/openeo-geotrellis/src/main/scala/org/openeo/geotrellis/file/FileRDDFactory.scala
+++ b/openeo-geotrellis/src/main/scala/org/openeo/geotrellis/file/FileRDDFactory.scala
@@ -16,7 +16,7 @@ import org.openeo.opensearch.OpenSearchResponses.{Feature, Link}
 import org.openeo.opensearch.backends.{CreodiasClient, OscarsClient}
 
 import java.net.URL
-import java.time.ZonedDateTime
+import java.time.{LocalTime, ZonedDateTime}
 import java.util
 import scala.collection.JavaConverters._
 
@@ -38,9 +38,13 @@ class FileRDDFactory(openSearch: OpenSearchClient, openSearchCollectionId: Strin
   private def getFeatures(boundingBox: ProjectedExtent, from: ZonedDateTime, to: ZonedDateTime, zoom: Int, sc: SparkContext): Seq[Feature] = {
     require(zoom >= 0)
 
+    val dateRange = // retain backwards compatibility for from == to
+      if (from isEqual to) (from `with` LocalTime.MIN, from `with` LocalTime.MAX)
+      else (from, to)
+
     val overlappingFeatures: Seq[Feature] = openSearch.getProducts(
       openSearchCollectionId,
-      Some((from, to)),
+      Some(dateRange),
       boundingBox,
       attributeValues = attributeValues.asScala.toMap,
       correlationId, ""

--- a/openeo-geotrellis/src/main/scala/org/openeo/geotrellis/file/FileRDDFactory.scala
+++ b/openeo-geotrellis/src/main/scala/org/openeo/geotrellis/file/FileRDDFactory.scala
@@ -40,7 +40,7 @@ class FileRDDFactory(openSearch: OpenSearchClient, openSearchCollectionId: Strin
 
     val overlappingFeatures: Seq[Feature] = openSearch.getProducts(
       openSearchCollectionId,
-      (from.toLocalDate, to.toLocalDate),
+      Some((from, to)),
       boundingBox,
       attributeValues = attributeValues.asScala.toMap,
       correlationId, ""

